### PR TITLE
[ECP-9277] Add Support for Billie payment method

### PR DESCRIPTION
**Pull request by SushmitaThakur**
This PR includes payment method class, handlers, dependency injection and other supporting structure for Billie payment method.

### CODE REVIEW

File: src/AdyenPaymentShopware6.php [hunk 1]
```diff
@@ -426,16 +426,24 @@ private function updateTo3100(UpdateContext $updateContext): void
 
     private function updateTo3150(UpdateContext $updateContext): void
```
Comment (candemiralp) [inline src/AdyenPaymentShopware6.php:427]: This payment method needs to be added with a separate function for the target version. For example, it's planned to be released as a part of v3.16.0, you should create a new update method in this class and call this method with a comparison. Version comparison examples can be found in the same class.

File: src/Handlers/BilliePaymentMethodHandler.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+{
+    public static bool $isOpenInvoice = true;
+    public static bool $supportsManualCapture = true;
+    public static bool $supportsPartialCapture = true;
+
+    public static function getPaymentMethodCode()
+    {
+        return 'klarna_b2b';
+    }
+}
```
Comment (candemiralp) [inline src/Handlers/BilliePaymentMethodHandler.php:37]: It's a best practice to add an single empty line to the end of the class.

File: src/PaymentMethods/BilliePaymentMethod.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getType(): string
+    {
+        return 'redirect';
+    }
+}
```
Comment (candemiralp) [inline src/PaymentMethods/BilliePaymentMethod.php:87]: It's a best practice to add an single empty line to the end of the class.

File: src/AdyenPaymentShopware6.php [hunk 4]
```diff
@@ -426,16 +426,24 @@ private function updateTo3100(UpdateContext $updateContext): void
...
+        foreach ([
             new PaymentMethods\GiftCardPaymentMethod(),
-            $updateContext->getContext()
-        );
-        $this->setPaymentMethodIsActive(
-            true,
-            $updateContext->getContext(),
-            new PaymentMethods\GiftcardPaymentMethod()
-        );
+            new PaymentMethods\BilliePaymentMethod()
```
Comment (candemiralp) [inline src/AdyenPaymentShopware6.php:435]: This payment method needs to be added with a separate function for the target version. For example, it's planned to be released as a part of v3.16.0, you should create a new update method in this class and call this method with a comparison. Version comparison examples can be found in the same class.

**Review by candemiralp — CHANGES_REQUESTED** [review]

**Review by khushboo-singhvi — DISMISSED** [review]

File: src/AdyenPaymentShopware6.php
Comment (khushboo-singhvi) [inline src/AdyenPaymentShopware6.php:1]: Hey, won't it be V4.x.x

File: src/AdyenPaymentShopware6.php [hunk 5]
```diff
@@ -461,6 +465,22 @@ private function updateTo3150(UpdateContext $updateContext): void
         }
     }
 
+    private function updateTo3160(UpdateContext $updateContext): void
```
Comment (khushboo-singhvi) [inline src/AdyenPaymentShopware6.php:468]: Same here?

**Review by candemiralp — APPROVED** [review]

**Pull request merged**